packages/preview-middleware-client/src/adp/controllers/BaseDialog.controller.ts

```diff
@@ -71,22 +72,26 @@ export default abstract class BaseDialog<T extends BaseDialogModel = BaseDialogM
         }
     }
 
-    /**
-     * Method is used in add fragment dialog controllers to get current control metadata which are needed on the dialog
-     * @returns control metadata and target aggregations
-     */
-    protected getControlMetadata(): { controlMetadata: ManagedObjectMetadata; targetAggregation: string[] } {
+    protected setRunTimeControl(): void {
         const selectorId = this.overlays.getId();
-
-        let controlMetadata: ManagedObjectMetadata;
-
         const overlayControl = getControlById(selectorId) as unknown as ElementOverlay;
         if (overlayControl) {
             this.runtimeControl = ControlUtils.getRuntimeControl(overlayControl);
-            controlMetadata = this.runtimeControl.getMetadata();
         } else {
             throw new Error('Cannot get overlay control');
         }
+    }
+
+    /**
+     * Method is used in add fragment dialog controllers to get current control metadata which are needed on the dialog
+     * @returns control metadata and target aggregations
+     */
+    protected getControlMetadata(): { controlMetadata: ManagedObjectMetadata; targetAggregation: string[] } {
+        this.setRunTimeControl();
```

Comment:
This looks quite error prone, because you need to always make sure it is called before using it. Better would be to create a getter which lazy loads the control.

Reply:
adapted